pandas/_libs/tslibs/util.pxd

```diff
@@ -236,13 +236,11 @@ cdef inline const char* get_c_string_buf_and_size(str py_string,
 
     Returns
     -------
-    buf : const char*
+    c_string_buf : const char*
     """
-    cdef:
-        const char *buf
-
-    buf = PyUnicode_AsUTF8AndSize(py_string, length)
-    return buf
+    if not py_string.isprintable():
```

Comment:
shouldn't we be calling the c-func here? this is a heavily used path

Reply:
Which function did you have in mind? The py_string is not something you can give to a C standard library function AFAIK. Plus this gets translated to C code calling python functions anyway.

Reply:
can you show the before & after generated c-code here.

Reply:
No longer relevant, I changed the code so that this is no longer there.